core/shared/src/main/scala/org/typelevel/log4cats/SamLogger.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import cats.*
+import cats.data.{EitherT, Kleisli, OptionT}
+
+/**
+ * A SAM-based Logger that extends LoggerKernel and provides a user-friendly interface. This is the
+ * new design that will eventually replace the current Logger trait.
+ */
+abstract class SamLogger[F[_], Ctx] extends LoggerKernel[F, Ctx] {
+
+  final def info(logBit: LogRecord[Ctx], others: LogRecord[Ctx]*)(implicit
```

Comment:
I would _strongly_ recommend against this API design. 

It's shiny, but very unfriendly to users, because it would allow stuff like this:

```
logger.info(caughtException, otherCaughtException)
```

This compiles and emits an `Info`-level log without a message, and only one of the two exceptions that were intended to be logged.